lib/fosdick/order.rb

```diff
@@ -198,7 +198,12 @@ def handle_error(response)
 
       case errors[0]
       when "Invalid"
-        raise InvalidError, errors.last
+        last_error = errors.last
+        if last_error.include?('Duplicate ExternalId')
```

Comment:
Fosdick is actually sending back `ExternalID`, not `ExternalId`, so this would not have worked in production.

Reply:
😬 

good catch 🔬